app/models/detector/suggested_resource.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: detector_suggested_resources
+#
+#  id          :integer          not null, primary key
+#  title       :string
+#  url         :string
+#  phrase      :string
+#  fingerprint :string
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+
+require 'stringex/core_ext'
+
+module Detector
+  # Detector::SuggestedResource stores custom hints that we want to send to the
+  # user in response to specific strings. For example, a search for "web of
+  # science" should be met with our custom login link to Web of Science via MIT.
+  class SuggestedResource < ApplicationRecord
+    before_save :update_fingerprint
+
+    def update_fingerprint
+      self.fingerprint = calculate_fingerprint
+    end
+
+    # This implements the OpenRefine fingerprinting algorithm. See
+    # https://openrefine.org/docs/technical-reference/clustering-in-depth#fingerprint
+    def calculate_fingerprint
```

Comment:
Slightly forward thinking and thus non-blocking, but having the `phrase` be an argument would allow this to be used on incoming search terms rather than only on instantiated `SuggestedResource`s.